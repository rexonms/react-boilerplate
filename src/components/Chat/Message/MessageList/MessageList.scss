.container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 50vw;
  max-height: 80vh;
  height: 100%;
  border-radius: 5px;
  background: rgba(255,255,255,.9);
  border: 1px solid #eee;
}
.image {
  width: 100%;
  height: 80%;
}
.bulb {
  width: 100%;
  height: 90%;
  margin-bottom: 10%;
  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
}
.bulbOn {
  background-image: url('./assets/bulbOn.svg')
}
.bulbOff {
  background-image: url('./assets/bulbOff.svg')
}
.button {
  margin: auto;
  padding: 10px 40px;
  background: #ccc;
  border: none;
  font-size: 14px;
  text-transform: uppercase;
  min-width: 150px;
  box-shadow: 0 0 black;
}
.button:focus {
  outline: none;
}
