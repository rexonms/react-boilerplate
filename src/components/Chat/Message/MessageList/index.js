import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './MessageList.scss';
import MessageItem from '../MessageItem/index';

const propTypes = {
  /** List of messages */
  list: PropTypes.arrayOf(PropTypes.shape({
    /** Unique id for the message */
    id: PropTypes.string.isRequired,
    /** Image URL for the avatar */
    avatarImageURL: PropTypes.string.isRequired,
    /** Recipient Name */
    recipientName: PropTypes.string.isRequired,
    /** Last message sent by recipient */
    lastMessage: PropTypes.string.isRequired,
  })).isRequired,
  /** Passing the onClick function to parent */
  onClick: PropTypes.func.isRequired,
};

const defaultProps = {
};

class MessageList extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClickHandler = this.onClickHandler.bind(this);
  }
  onClickHandler(email) {
    this.props.onClick(email);
  }
  render() {
    return (
      <div className={styles.container}>
        {this.props.list.map(item => (
          <MessageItem
            key={item.id}
            avatarImageURL={item.avatarImageURL}
            recipientName={item.recipientName}
            lastMessage={item.lastMessage}
            email={item.email}
            imageAltText={item.imageAltText}
            onClick={this.onClickHandler}
          />
        ))}
      </div>
    );
  }
}

MessageList.defaultProps = defaultProps;
MessageList.propTypes = propTypes;

export default MessageList;
