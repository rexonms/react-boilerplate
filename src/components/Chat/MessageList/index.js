import React from 'react';
import PropTypes from 'prop-types';

import styles from './MessageList.scss';
import MessageItem from '../MessageItem/index';

const MessageList = props => (
  <div className={`${styles.container}`}>
    {props.list.map( item => (
      <MessageItem
        key={item.id}
        avatarImageURL={item.avatarImageURL}
        recipientName={item.recipientName}
        lastMessage={item.lastMessage}
      />)
    )}
  </div>
);

MessageList.defaultProps = {};

MessageList.propTypes = {
  /** And array of object that has message list */
  list: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    avatarImageURL: PropTypes.string.isRequired,
    recipientName: PropTypes.string.isRequired,
    lastMessage: PropTypes.string.isRequired,
  })).isRequired,
};

export default MessageList;
