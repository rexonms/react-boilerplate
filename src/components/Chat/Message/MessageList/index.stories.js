import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import MessageList from './index';

const stories = storiesOf('components/Chat/Message/MessageList', module);
export const props = {
  list: [
    {
      id: '1',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/111111/eeeeee',
      imageAltText: 'John Smith',
      recipientName: 'John Smith',
      lastMessage: 'Where have you been?',
      email: 'john.smith@gmail.com',
    },
    {
      id: '2',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/333333/eeeeee',
      imageAltText: 'Jenny Smith',
      recipientName: 'Jenny',
      lastMessage: 'And then there this time when I was thing of lots of stuff. A single line field is used when the input anticipated by the CHAT_BOX_USER is a single line of text as opposed to a paragraph.',
      email: 'Jenny.smith@gmail.com',
    },
    {
      id: '3',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/cccccc/eeeeee',
      imageAltText: 'Jhonothan Smith',
      recipientName: 'Mr. Jhonothan Goldsmith',
      lastMessage: 'yt',
      email: 'Jhonothan.smith@gmail.com',
    },
    {
      id: '4',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/aaaaaa/eeeeee',
      imageAltText: '',
      recipientName: '',
      lastMessage: 'hola',
      email: 'smith@gmail.com',
    },
  ],
};


stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('MessageList', withInfo(`
  A Stateful Component example that toggles the bulb state based on CHAT_BOX_USER click event.
~~~js
~~~
`)(() => {
return (
  <MessageList
   list={props.list}
   onClick={action(this.props.email)}
  />
);
}));
