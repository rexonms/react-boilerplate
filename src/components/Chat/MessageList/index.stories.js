import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import MessageList from './index';

const stories = storiesOf('Components/Chat/MessageList', module);
stories.addDecorator(withKnobs);

// Knobs as dynamic variables.
stories.addDecorator(story => (story()));

stories.add('default', withInfo(`
  Displays list of all the messages
  ~~~js
    <MessageList list={list} />
  ~~~
`)(() => {
  const messages = [
    {
      id: '1',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/111111/eeeeee',
      recipientName: 'John Smith',
      lastMessage: 'Where have you been?',
    },
    {
      id: '2',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/333333/eeeeee',
      recipientName: 'Jenny',
      lastMessage: 'And then there this time when I was thing of lots of stuff. A single line field is used when the input anticipated by the user is a single line of text as opposed to a paragraph.',
    },
    {
      id: '3',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/cccccc/eeeeee',
      recipientName: 'Mr. Jhonothan Goldsmith',
      lastMessage: 'yt',
    },
    {
      id: '4',
      avatarImageURL: 'http://dummyimage.com/mediumrectangle/aaaaaa/eeeeee',
      recipientName: '',
      lastMessage: 'hola',
    }
  ]
  return (
    <MessageList list={messages} />
  )
}));

