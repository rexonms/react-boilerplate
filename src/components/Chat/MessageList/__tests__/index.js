import React from 'react';
import renderer from 'react-test-renderer';

import MessageList from '../index';
const messages = [
  {
    id: '1',
    avatarImageURL: 'http://dummyimage.com/mediumrectangle/111111/eeeeee',
    recipientName: 'John Smith',
    lastMessage: 'Where have you been?',
  },
]
it('renders correctly', () => {
  const tree = renderer.create(
    <MessageList list={messages} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
