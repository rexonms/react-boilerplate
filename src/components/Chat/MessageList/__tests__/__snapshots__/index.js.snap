// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div
      className="avatar"
    >
      <div
        className="container"
      >
        <img
          src="http://dummyimage.com/mediumrectangle/111111/eeeeee"
        />
      </div>
    </div>
    <div
      className="details"
    >
      <h6
        className="default"
      >
        John Smith
      </h6>
      <p
        className="small "
      >
        Where have you been?
      </p>
    </div>
  </div>
</div>
`;
